### DISCUSSION TITLE
Wildcards and missing entities

### DISCUSSION THREAD

**Discussion by pvandyken** [khanlab/snakebids#186] [General]
Should the presence of a wildcard require that the wildcard be present? For example, if `run` is a wildcard, should paths without the `run` entity be collected (and `run` would just be excluded from the zip lists, etc). Or put another way, does having `run` as a wildcard imply a filter `run=Query.ANY`?

**Comment by tkkuehn** [comment]
My gut reaction: No, mainly because of the (I think) common situation where you want to support the subject and session wildcards if they're there but don't actually care whether there are sessions or not. I think this is the current wildcard behaviour and it would be good to make sure that's explicit.

Basically in my opinion we want to support each of the three following possibilities:

- The entity must not be defined (Filter `Query.NONE`; or implicitly, neither a wildcard nor a filter)
- The entity must be captured in a wildcard if defined but may be absent (Wildcard)
- The entity must be defined and captured in a wildcard (Filter `Query.ANY`).

**Reply by pvandyken** [reply]
Sounds good. This is the current behaviour, but I never thought about it until now setting up support for `Query.None` and `Query.ANY`. I wanted to make sure that this design was intentional
